android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BeaconConfigFragment.java

```diff
@@ -327,21 +335,28 @@ private void saveEditCardUrlToBeacon() {
     if (!URLUtil.isNetworkUrl(url)) {
       url = "http://" + url;
     }
+    // Create the callback object to set the url
+    UrlShortener.ModifiedUrlCallback urlSetter = new UrlShortener.ModifiedUrlCallback() {
+      public void onNewUrl(String newUrl){
+        // Write the url to the device
+        try {
+          // Note: setting txPower here only really updates txPower for v1 beacons
+          ConfigUriBeacon configUriBeacon = new ConfigUriBeacon.Builder()
+              .txPowerLevel(TX_POWER_DEFAULT)
+              .uriString(newUrl)
+              .build();
+          mUriBeaconConfig.writeUriBeacon(configUriBeacon);
+        } catch (URISyntaxException e) {
+          e.printStackTrace();
+          Toast.makeText(getActivity(), R.string.config_url_error, Toast.LENGTH_SHORT).show();
```

Comment:
I'm struggling to find a decent pattern for doing this.  There could be multiple of these that need to be canceled.  I could add them all into an arraylist and iterate through and cancel.  If I wanted to keep that list clean, I would have to check it for finished tasks before adding a new one or I could remove the items from the list when the task completes, but then I'd need to use a synchronized list.  Do you have any suggestions on the best way to go about this?